service-search-nls/src/main/java/fi/nls/oskari/search/channel/ELFGeoLocatorSearchChannel.java

```diff
@@ -428,35 +430,50 @@ public void calculateCommonFields(final SearchResultItem item) {
         if(PROPERTY_AUTOCOMPLETE_URL == null || PROPERTY_AUTOCOMPLETE_URL.isEmpty()) {
             return Collections.emptyList();
         }
+        JSONObject jsonObject;
         try {
             log.info("Creating autocomplete search url with url:", PROPERTY_AUTOCOMPLETE_URL);
             HttpURLConnection conn = IOHelper.getConnection(PROPERTY_AUTOCOMPLETE_URL,
                     PROPERTY_AUTOCOMPLETE_USERNAME, PROPERTY_AUTOCOMPLETE_PASSWORD);
             IOHelper.writeToConnection(conn, getElasticQuery(searchString));
             String result = IOHelper.readString(conn);
-            JSONObject jsonObject = new JSONObject(result);
-
-            JSONArray jsonHitsArray = jsonObject.getJSONObject("hits").getJSONArray("hits");
-            List<String> resultList = new ArrayList<>();
-            for (int i = 0; i < jsonHitsArray.length(); ++i ) {
-                String resultName = jsonHitsArray.getJSONObject(i).getJSONObject("_source").getString("name");
-                resultList.add(resultName);
-            }
-            return resultList;
+            jsonObject = new JSONObject(result);
         }
         catch (Exception ex) {
             log.error("Couldn't open or read from connection for search channel!");
             throw new RuntimeException("Couldn't open or read from connection!", ex);
```

Comment:
It usually isn't a good idea to log AND throw an exception as it tends to lead to massive amounts of logging for an issue. I'll modify this a bit so we can include it in the next release.